packages/core/integration-tests/test/sourcemaps.js

```diff
@@ -314,4 +323,159 @@ describe('sourcemaps', function() {
+      {
+        source: '../integration/sourcemap-css/style.css',
+        name: null,
+        line: indexToLineCol(input, input.indexOf('body')).line,
+        column: indexToLineCol(input, input.indexOf('body')).column
+      },
+      "map 'body'"
+    );
+
+    // assert.deepStrictEqual(
+    //   consumer.originalPositionFor(indexToLineCol(raw, raw.indexOf('{'))),
+    //   {
+    //     source: '../integration/sourcemap-css/style.css',
+    //     name: null,
+    //     ...indexToLineCol(input, input.indexOf('{'))
+    //   },
+    //   "map '{'"
+    // );
+
+    assert.deepStrictEqual(
+      consumer.originalPositionFor(
+        indexToLineCol(raw, raw.indexOf('background-color'))
+      ),
+      {
+        source: '../integration/sourcemap-css/style.css',
+        name: null,
+        line: indexToLineCol(input, input.indexOf('background-color')).line,
+        column: indexToLineCol(input, input.indexOf('background-color')).column
+      },
+      "map 'background-color'"
+    );
+
+    // assert.deepStrictEqual(
+    //   consumer.originalPositionFor(indexToLineCol(raw, raw.indexOf('}'))),
+    //   {
+    //     source: '../integration/sourcemap-css/style.css',
+    //     name: null,
+    //     ...indexToLineCol(input, input.indexOf('}'))
+    //   },
+    //   "map '}'"
+    // );
+  });
+
+  it('should create a valid sourcemap for a CSS bundle with imports', async function() {
+    let b = await bundle(
+      path.join(__dirname, '/integration/sourcemap-css-import/style.css'),
+      {minify: true}
+    );
+
+    await assertBundleTree(b, {
+      name: 'style.css',
+      assets: ['style.css', 'other-style.css'],
+      childBundles: [
+        {
+          name: 'style.css.map',
+          type: 'map'
+        }
+      ]
+    });
+
+    let style = (await fs.readFile(
+      path.join(__dirname, '/integration/sourcemap-css-import/style.css')
+    )).toString();
+    let otherStyle = (await fs.readFile(
+      path.join(__dirname, '/integration/sourcemap-css-import/other-style.css')
+    )).toString();
+    let raw = (await fs.readFile(
+      path.join(__dirname, '/dist/style.css')
+    )).toString();
+    let map = (await fs.readFile(
+      path.join(__dirname, '/dist/style.css.map')
+    )).toString();
+
+    assert(raw.includes('/*# sourceMappingURL=/style.css.map*/'));
+
+    let consumer = await new SourceMapConsumer(map);
+
+    // assert.deepStrictEqual(
+    //   consumer.originalPositionFor(indexToLineCol(raw, raw.indexOf('body'))),
+    //   {
+    //     source: '../integration/sourcemap-css-import/style.css',
+    //     name: null,
+    //     ...indexToLineCol(style, style.indexOf('body'))
+    //   },
+    //   "map 'body'"
+    // );
+
+    assert.deepStrictEqual(
+      consumer.originalPositionFor(
+        indexToLineCol(raw, raw.indexOf('background-color'))
+      ),
+      {
+        source: '../integration/sourcemap-css-import/style.css',
```

Comment:
I have some thoughts... maybe some of them not entirely correct, but would share them.

Surprisingly, this one fails for me with

```
 + expected - actual

 -  "source": "../integration/sourcemap-css-import/other-style.css"
 +  "source": "../integration/sourcemap-css-import/style.css"
```

Does it pass for you? Luckily, the source maps actually work! It's the tests that fail for some reason.

I think the reason is the assumption how `consumer.originalPositionFor` works and the implementation of `indexToLineCol`. As far as I was able to reproduce, `originalPositionFor` takes into consideration the contents of each `@import`. Meaning, if we look for `originalPosition` for a certain expression, the result returned would also count lines in all `@import`s in the file where the expression has been found. This is important, because it means that 

```
consumer.originalPositionFor(
  indexToLineCol(raw, raw.indexOf('background-color'))
)
```

would consider lines in `other-style.css` too, however we only match it against the contents of `style.css` 

```
line: indexToLineCol(style, style.indexOf('background-color')).line
```

As opposed to `style.css` concatenated with `other-style.css.` To verify: try adding lines / styles in `other-style.css` and see how `originalPositionFor` changes its result (lines increase). 

So I think a small adjustment to consider all imports lines when doing assertion would fix this.

For instance,

```
 line: indexToLineCol(style, style.indexOf('background-color')).line,
```

would be (more or less, we have to take into consideration extracting number of @import lines I guess)

```
 line: indexToLineCol(style.concat(otherStyle), style.concat(otherStyle).indexOf('background-color')).line,
```

Maybe the function `indexToLineCol` has to be then rewritten according to how `originalPositionFor works` and this would be it. 

I'm not entirely sure, because I don't have complete understanding of what's going on here, but I guess this would be my bet why tests fail.